packages/block-editor/src/components/inserter/utils.js

```diff
@@ -1,45 +1,71 @@
+// TODO jsdoc + tests
+const getVariationMapper = ( item ) => ( variation ) => ( {
```

Comment:
I used to be a huge fan of currying and other FP techniques in JS. :) In the context of Gutenberg, I'm a bit more conservative. Plus, arrow functions mitigate some of the need.

1. Anyway, the name is probably what needs the most work. `getItemFromVariation`? wdyt?

2. As for `( item ) => ( variation ) =>` vs. `( item, variation ) =>`, it's ultimately your preference.